### PR TITLE
Increase coverage

### PR DESCRIPTION
**Pull request by rebkwok**
Add tests and remove some unused code for the easy wins to increase coverage.  Coverage is now up to 99.73%, and the remaining uncovered code is mostly in query engines - I think some of those are probably OK to mark as no-cover, but leaving them for people who know more about it than I do
```
Name                                         Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------------------------
databuilder/query_engines/base_sql.py          231      2     62      0    99%   108, 235
databuilder/query_engines/mssql_dialect.py      33      1      8      1    95%   22
databuilder/query_engines/mssql_lib.py         140      5     28      3    95%   222->exit, 249, 297, 313-315
databuilder/query_engines/spark_dialect.py      50      0      2      1    98%   248->250
tests/lib/databases.py                          79      1     12      0    99%   75
----------------------------------------------------------------------------------------
TOTAL                                         4524      9    622      5    99%
```

### CODE REVIEW

File: tests/test_functools_utils.py [hunk 1]
```diff
@@ -30,3 +32,6 @@ def method_c(self, value):
     assert test_obj.method(A()) == "a_or_b"
     assert test_obj.method(B()) == "a_or_b"
     assert test_obj.method(C()) == "c"
+    # can't
```
Comment (ghickman) [inline tests/test_functools_utils.py:35]: can't?

**Review by ghickman — APPROVED** [review]
🎉

File: tests/test_functools_utils.py (hunk 1 above)
Comment (rebkwok) [reply tests/test_functools_utils.py:35]: no idea :rofl: 
I imagine I had a thought.

Comment (rebkwok) [reply tests/test_functools_utils.py]: Half a thought.  I've completed it now :smile:

**Pull request merged**